src/boxed.rs

```diff
@@ -56,4 +84,13 @@ mod tests {
     //     let ptr = unsafe { alloc::alloc::alloc(layout) };
     //     assert!(ptr.is_null());
     // }
+
+    /// Zero-sized types support requires the allocator API.
+    #[test]
+    fn trybox_zst() {
+        #[cfg(feature = "unstable")]
+        assert!(Box::try_new(()).is_ok());
+        #[cfg(not(feature = "unstable"))]
+        assert!(Box::try_new(()).is_err());
+    }
```

Comment:
Sounds good, I didn't even think about that option. I believe it should be sound to have a dangling pointer to a ZST since there's no way it would be dereferenced. The only bit in reading [the `NonNull` docs](https://doc.rust-lang.org/stable/core/ptr/struct.NonNull.html) that gives me pause is:
> Unlike `*mut T`, `NonNull<T>` is covariant over `T`. If this is incorrect for your use case, you should include some `PhantomData` in your type to provide invariance

I _think_ this is fine being covariant, but I need to refresh my memory about variance rules to be sure. I'll submit an updated PR in the meantime, and if you're confident about it, I'll defer to you.

Reply:
sounds fine.